src/scripts/main.js

```diff
@@ -1,3 +1,236 @@
+  restart.style.display = 'block';
+});
+
+restart.addEventListener('click', () => {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      board[r][c] = 0;
+
+      const tile = document.getElementById(r.toString() + '-' + c.toString());
+
+      tile.innerText = '';
+      tile.classList = 'tile';
+    }
+  }
+
+  score = 0;
+  document.querySelector('.game-score').innerText = score;
+  start.style.display = 'block';
+  restart.style.display = 'none';
+  messageLose.classList.add('hidden');
+  messageStart.style.display = 'block';
+});
+
+window.onload = function() {
+  setGame();
+};
+
+function setGame() {
+  board = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      const tile = document.createElement('div');
+
+      tile.id = r.toString() + '-' + c.toString();
+
+      const num = board[r][c];
+
+      updateTile(tile, num);
+
+      document.querySelector('.board').append(tile);
+    }
+  }
+}
+
+function hasEmptyTile() {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      if (board[r][c] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function setTwo() {
+  if (!hasEmptyTile()) {
+    messageLose.classList.remove('hidden');
+
+    return;
+  }
+
+  let found = false;
+
+  while (!found) {
+    const r = Math.floor(Math.random() * rows);
+    const c = Math.floor(Math.random() * columns);
+
+    if (board[r][c] === 0) {
+      board[r][c] = 2;
+
+      const tile = document.getElementById(r.toString() + '-' + c.toString());
+
+      tile.innerText = '2';
+      tile.classList.add('tile--2');
+      found = true;
+    }
+  }
+}
+
+function updateTile(tile, num) {
+  tile.innerText = '';
+  tile.classList.value = '';
+  tile.classList.add('tile');
+
+  if (num > 0) {
+    tile.innerText = num;
+
+    if (num <= 2048) {
+      tile.classList.add('tile--' + num.toString());
+    }
+
+    if (num === 2048) {
+      messageWin.classList.remove('hidden');
+    }
+  }
+}
+
+document.addEventListener('keyup', (e) => {
+  if (e.code === 'ArrowLeft') {
+    slideLeft();
+    setTwo();
+  } else if (e.code === 'ArrowRight') {
+    slideRight();
+    setTwo();
+  } else if (e.code === 'ArrowUp') {
+    slideUp();
+    setTwo();
+  } else if (e.code === 'ArrowDown') {
+    slideDown();
+    setTwo();
+  }
+  document.querySelector('.game-score').innerText = score;
+});
+
+function filterZero(row) {
+  return row.filter((num) => num !== 0);
+}
+
+function slide(row) {
+  let Row = row;
+
+  Row = filterZero(Row);
+
+  for (let i = 0; i < Row.length; i++) {
+    if (Row[i] === Row[i + 1]) {
+      Row[i] *= 2;
+      Row[i + 1] = 0;
+      score += Row[i];
+    }
+  }
+
+  Row = filterZero(Row);
+
+  while (Row.length < columns) {
+    Row.push(0);
+  }
+
+  return Row;
+}
+
+function slideLeft() {
+  for (let r = 0; r < rows; r++) {
```

Comment:
use more meaningful names for those variables and fix it everywhere
```suggestion
  for (let i = 0; i < rows; i++) {
```